From 35dff4ab509183fce64494b087a3452349814a9e Mon Sep 17 00:00:00 2001
From: Vladimir Oltean <vladimir.oltean@nxp.com>
Date: Wed, 20 Jun 2018 17:43:52 +0300
Subject: [PATCH] sdk_dpaa: dpa_ioctl: allow passing of SIOCSMIIREG and
 SIOCGMIIPHY to phylib

* Previously the sdk_dpaa driver only supported SIOCGMIIREG (Read PHY register
  over MDIO). The extra ioctl calls are for writing PHY registers over
  MDIO (SIOCSMIIREG) and getting the address of the net device's phydev
  (SIOCGMIIPHY).
* This enables programs that use the extra ioctl calls, such as "phytool write",
  to not end up failing with -EINVAL in the dpa_ioctl() function.

Signed-off-by: Vladimir Oltean <vladimir.oltean@nxp.com>
---
 drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_common.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_common.c b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_common.c
index bbaf29c..a8132ff 100644
--- a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_common.c
+++ b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_common.c
@@ -474,7 +474,7 @@ int dpa_ioctl(struct net_device *dev, struct ifreq *rq, int cmd)
 	if (!netif_running(dev))
 		return -EINVAL;
 
-	if (cmd == SIOCGMIIREG) {
+	if (cmd == SIOCGMIIREG || cmd == SIOCSMIIREG || cmd == SIOCGMIIPHY) {
 		if (!dev->phydev)
 			ret = -EINVAL;
 		else
-- 
2.7.4

